nixos/tests/dolibarr.nix

```diff
@@ -0,0 +1,113 @@
+    environment.systemPackages = let
+      auto-install-dolibarr = pkgs.writers.writePython3Bin "auto-install-dolibarr" {
+        libraries = [ pkgs.python3Packages.selenium ];
+      } ''
+        from selenium.webdriver import Firefox
+        from selenium.webdriver.firefox.options import Options
+        from selenium.webdriver.support.wait import WebDriverWait
+        from selenium.webdriver.support import expected_conditions as EC
+        from selenium.webdriver.common.by import By
+
+        options = Options()
+        options.add_argument('--headless')
+        print("Starting...")
+        driver = Firefox(options=options)
+
+        print("Opening http://localhost/install...")
+        driver.get('http://localhost/install/')
+        print("Starting sequence...")
+        driver.find_element(
+          'xpath',
+          '/html/body/form/table/tbody/tr/td/table/tbody/' +
+          'tr/td/table/tbody/tr[22]/td[3]/a'
+        ).click()
+        tmp = driver.find_element('xpath', '//*[@id="main_data_dir"]')
+        tmp.clear()
+        tmp.send_keys("/var/lib/dolibarr/documents")
+
+        tmp = driver.find_element('xpath', '//*[@id="db_pass"]')
+        tmp.clear()
+        tmp.send_keys('${dbPassword}')
+
+        driver.find_element('xpath', '/html/body/form/div/input').click()
+        driver.find_element('xpath', '/html/body/form/div[1]/input').click()
+
+        print("Waiting for installation to complete...")
+        WebDriverWait(driver, 60).until(
+          EC.element_to_be_clickable((By.XPATH, "/html/body/form/div/input"))
+        )
+
+        print("Configuring the admin credentials...")
+        driver.find_element('xpath', '/html/body/form/div/input').click()
+
+        driver.find_element('xpath', '/html/body/form/div/input').click()
+
+        tmp = driver.find_element('xpath', '//*[@id="login"]')
+        tmp.clear()
+        tmp.send_keys('admin')
+
+        tmp = driver.find_element('xpath', '//*[@id="pass"]')
+        tmp.clear()
+        tmp.send_keys('password')
+
+        tmp = driver.find_element('xpath', '//*[@id="pass_verif"]')
+        tmp.clear()
+        tmp.send_keys('password')
+
+        driver.find_element('xpath', '/html/body/form/div/input').click()
+
+        driver.find_element(
+          'xpath',
+          '/html/body/form/table/tbody/tr/td/table/tbody/tr/td/div[2]/a'
+        ).click()
+
+        print("Done")
+      '';
+    in [ pkgs.firefox-unwrapped pkgs.geckodriver auto-install-dolibarr ];
+
+    services.dolibarr = {
+      enable = true;
+      domain = "localhost";
+      preInstalled = false;
+      initialDbPasswordFile = passwordFile;
+    };
+  };
+
+  testScript = let
+    mysqlCreds = pkgs.writeText "mysql-creds" ''
+      [client]
+      user=dolibarr
+      password=${dbPassword}
+    '';
+  in ''
+    start_all()
+
+    machine.wait_for_unit("multi-user.target")
+    machine.wait_for_unit("nginx.service")
+    machine.wait_for_unit("mysql.service")
```

Comment:
```suggestion
```

This shouldn't be necessary if dolibarr sets its systemd options correct and we miss a testing potential here.